python_modules/dagster/dagster_tests/core_tests/test_logging.py

```diff
@@ -2,20 +2,65 @@
 import logging
 import re
 
+from contextlib import contextmanager
+
+from dagster import check
+from dagster.core.definitions import PipelineDefinition
 from dagster.core.events import DagsterEvent
+from dagster.core.init_context import InitLoggerContext
 from dagster.core.execution_plan.objects import StepFailureData
-from dagster.core.log import DagsterLog, DAGSTER_DEFAULT_LOGGER
+from dagster.core.log_manager import DagsterLogManager
+from dagster.core.loggers import colored_console_logger
 from dagster.utils.error import SerializableErrorInfo
 
 
 REGEX_UUID = r'[a-z-0-9]{8}\-[a-z-0-9]{4}\-[a-z-0-9]{4}\-[a-z-0-9]{4}\-[a-z-0-9]{12}'
 REGEX_TS = r'\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d{6}'
 
+DAGSTER_DEFAULT_LOGGER = 'dagster'
+
+
+@contextmanager
+def _setup_logger(name, log_levels=None, register_levels=True):
+    '''Test helper that creates a new logger.
+
+    Args:
+        name (str): The name of the logger.
+        log_levels (Optional[Dict[str, int]]): Any non-standard log levels to expose on the logger
+            (e.g., logger.success)
+        register_levels (bool): Whether to register non-standard log levels with the Python logging
+            module. Default True.
+
+    '''
+    log_levels = check.opt_dict_param(log_levels, 'log_levels')
+
+    def add_log_level(value, name):
```

Comment:
this section is kind of terrifying. at minimum a big 'ole comment block would be good explaining what is going on and why public APIs are not sufficient

Reply:
Yep